src/openrct2/scripting/ScRide.hpp

```diff
@@ -642,6 +641,21 @@ namespace OpenRCT2::Scripting
             }
         }
 
+        uint8_t downtime_get() const
+        {
+            auto ride = GetRide();
+            return ride != nullptr ? ride->downtime : 0;
+        }
+        void downtime_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto ride = GetRide();
+            if (ride != nullptr)
+            {
+                ride->downtime = std::min<uint8_t>(value, 100);
```

Comment:
This value will be overwritten the next time `ride_breakdown_update` calculates its percentage based on the downtime history. Exposing the history to plugins and allowing modifications instead of setting a value that gets recalculated would be preferable.

Reply:
@Basssiiie not all properties need to have setters, or at least they could be added later if it isn't important right now.

Reply:
Ah didn't realize that it gets overridden. I'll remove this setter.